src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/Build/PublishItemsOutputGroupProvider.cs

```diff
@@ -24,11 +24,11 @@ internal class PublishItemsOutputGroupProvider : IOutputGroupProvider
         private readonly ConfiguredProject _configuredProject;
 
         [ImportingConstructor]
-        internal PublishItemsOutputGroupProvider(IProjectAccessor projectAccessor, ConfiguredProject configuredProject)
+        internal PublishItemsOutputGroupProvider(IProjectAccessor projectAccessor, ConfiguredProject configuredProject, JoinableTaskFactory joinableTaskFactory)
         {
             _projectAccessor = projectAccessor;
             _configuredProject = configuredProject;
-            _outputGroups = new AsyncLazy<IImmutableSet<IOutputGroup>>(GetOutputGroupMetadataAsync);
+            _outputGroups = new AsyncLazy<IImmutableSet<IOutputGroup>>(GetOutputGroupMetadataAsync, joinableTaskFactory);
```

Comment:
@davkean please review this change.

Reply:
This isn't correct and won't run. There is no importable JoinableTaskFactory, it needs to import IProjectThreadingService.

Reply:
Fixed.